### PR TITLE
Add Tekton Task and TaskRun to create and delete Ingress

### PR DESCRIPTION
**Pull request by a-roberts**
This is for https://github.com/tektoncd/experimental/issues/242

# Changes

Again largely based on the work by @akihikokuroda [here](https://github.com/tektoncd/triggers/blob/master/docs/create-webhook.yaml)

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: webhooks-extension/config/triggers_prototype/ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            - $(inputs.params.ExternalUrl)
+            secretName: $(inputs.params.CertificateSecretName)
+          rules:
+          - host: $(inputs.params.ExternalUrl)
+            http:
+              paths:
+              - backend:
+                  serviceName: $(inputs.params.EventListenerName)
+                  servicePort: 8082
+      EOF
```
Comment (akihikokuroda) [inline webhooks-extension/config/triggers_prototype/ingress.yaml]: Is this indentation correct?

Comment (a-roberts) [reply webhooks-extension/config/triggers_prototype/ingress.yaml]: Without it we get

```
Handling Ingress
sh: syntax error: unexpected end of file (expecting "fi")
```

So it must be, I didn't think it was either

Comment (akihikokuroda) [reply webhooks-extension/config/triggers_prototype/ingress.yaml]: OK.  I thought it must be same level as "cat".

File: webhooks-extension/config/triggers_prototype/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Note that in order for the EventListener service to be reachable over Ingress, you should install your own LoadBalancer - for example with:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/mandatory.yaml
+
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/provider/cloud-generic.yaml
+```
+
+To test the Ingress set up you can then do the curl:
```
Comment (dibbles) [inline webhooks-extension/config/triggers_prototype/README.md]: the curl ..... "a curl" ...... or better yet "use the curl command"

File: webhooks-extension/config/triggers_prototype/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      description: The secret name for the Ingress certificate
+      default: "secret1"
+    - name: ExternalUrl 
+      description: The external access URL
+      default: "https://$(inputs.params.EventListenerName).replaceme.nip.io"
+    - name: EventListenerName 
+      description: The EventListener name
+      default: "listener"
+    - name: Namespace
+      description: "The namespace to use for where the secret will be created/deleted"
```
Comment (dibbles) [inline webhooks-extension/config/triggers_prototype/ingress.yaml]: "The namespace where the certificate secret will be created/deleted"

File: webhooks-extension/config/triggers_prototype/ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      default: "tekton-pipelines"
+  steps:
+  - name: generate-certificate
+    image: frapsoft/openssl
```
Comment (dibbles) [inline webhooks-extension/config/triggers_prototype/ingress.yaml]: Probably need to check licenses on the images being used

File: webhooks-extension/config/triggers_prototype/ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+          - hosts:
+            - $(inputs.params.ExternalUrl)
+            secretName: $(inputs.params.CertificateSecretName)
+          rules:
+          - host: $(inputs.params.ExternalUrl)
+            http:
+              paths:
+              - backend:
+                  serviceName: $(inputs.params.EventListenerName)
+                  servicePort: 8082
```
Comment (dibbles) [inline webhooks-extension/config/triggers_prototype/ingress.yaml]: What about having the port as an input?  I think it is supposed to be changing to port 80 to support knative at some point.

File: webhooks-extension/config/triggers_prototype/ingress.yaml (hunk 4 above)
Comment (a-roberts) [reply webhooks-extension/config/triggers_prototype/ingress.yaml]: Pulled and ran interactively with
`docker run -it --entrypoint /bin/ash frapsoft/openssl`

No license file found in here
openssl binary itself, which we know is in here (obv) is version 1.0.2 from September 2016 (I hate how old this image is, but not enough that I want us to maintain our own openssl image)

```
For any release made before OpenSSL 3.0.0 (namely the 1.1.1, 1.1.0, 1.0.2, and all prior releases including those not currently supported), the dual OpenSSL and SSLeay license applies. Note that this is also true for any updates to those releases -- the "letter suffix" -- no matter when they are made. It also applies to the git branches for all those releases, and to any public forks that have not rebased to master (or 3.0.0).
```

Also, I remember doing export regulation training, openssl is definitely something you can use for encryption. Sooo we could recommend running a script we provide, assuming they've got openssl, then mount in as a secret, and have that certificate used by this task.

Comment (a-roberts) [reply webhooks-extension/config/triggers_prototype/ingress.yaml]: Alternative approach which I lovingly refer to as BYOC at https://github.com/tektoncd/experimental/pull/276

**Comment by dibbles** [comment]
Closing as superceded by #276

**Pull request closed without merging**